### ISSUE TITLE
soundStream constructor is being called twice

### ISSUE THREAD

**Issue by dimitre** [openframeworks/openFrameworks#7295]
ofSoundStream::ofSoundStream() is being called twice, even if in my example I'm only using one ofSoundStream 
This calls this function twice setSoundStream(std::make_shared<OF_SOUND_STREAM_TYPE>());
which creates two different objects in memory
both for iOS and macOS. probably all platforms

it can be tested in a simple example, only creating one ofSoundStream object
```c++
class ofApp : public ofBaseApp{
	public:
		void setup();
		void update();
		void draw();
		void keyPressed(int key);

	ofSoundStream soundStream;
};
```

To inspect what is being called I've made this alterations
```c++
using std::cout;
using std::endl;
//------------------------------------------------------------
ofSoundStream::ofSoundStream(){
	cout << "ofSoundStream" << endl;
	#ifdef OF_SOUND_STREAM_TYPE
		setSoundStream(std::make_shared<OF_SOUND_STREAM_TYPE>());
	#endif
}

//------------------------------------------------------------
void ofSoundStream::setSoundStream(const shared_ptr<ofBaseSoundStream> & soundStreamPtr){
    std::cout << "setSoundStream " << &soundStreamPtr << std::endl;
	soundStream = soundStreamPtr;
}

```

result
```
ofSoundStream
setSoundStream 0x16d62e230
2023-02-02 21:15:20.708128-0300 Empty[47621:2755547] Metal API Validation Enabled
ofSoundStream
setSoundStream 0x16d6331d0

```

**Comment by artificiel** [comment]
there is a global soundStream tucked in an anonymous namespace in ofSoundStream.cpp 

https://github.com/openframeworks/openFrameworks/blob/72a6ea3296b554093b6c168da4c1a7eb13d4e05d/libs/openFrameworks/sound/ofSoundStream.cpp#L23-L25

it connects with the pseudo-static `ofSoundStreamSetup|Start|Stop|Close()`).

so having `ofSoundStream soundStream;` in app.h is redundant as it will simply "overwrite" the global systemSoundStream and lay around doing nothing (but provide trampoline to the global's methods). I think this stems from a design goal of OF to encourage the pattern of having plain objects declared in .h as it seems there was an effort done to make it behave like a standard allocation.

so in short: remove the `ofSoundStream soundStream;` in app.h and the example works as well, but you cannot make use of the direct soundStream.functions().

maybe the idea of doing that that should not be proposed, and if there are more methods that need access create the desired global ofSoundStream*() so there is a consistant access pattern.

**Comment by dimitre** [comment]
Yeah this can explain, but I'm not sure why it only allocates the other object once we have declared ofSoundStream soundStream; and at the same time. 

I've noticed some years ago using ofSoundStream object was preferred over ofSoundStreamSetup. Maybe it it time to deprecate ofSoundStreamSetup ?

**Comment by artificiel** [comment]
as my understanding of compile units goes: when an element of the ofSoundStream.h gets declared (as in the .h, or by using ofSoundStreamSetup()), the anonymous namespace gets linked/allocated, hence the 1st instanciation as part of the global ofBaseApp. 

then the actual ofApp.h declaration gets instanciated. 

so there is an ofSoundStream in the global systemSoundStream shared_ptr, and an ofSoundStream in the .h.... the ofSoundStream*() methods interact with the global, while the soundStream.methods() interact with the specific .h instance. in the case of iOS, both instances will eventually be talking to the same [AVAudioSession sharedInstance] so it look like a buggy design to allow multiple instances (even more so if one of the instances is "hidden").

I don't know which is "better", but considering the singleton-ish properties of ofSoundStream, it seems the global approach might be simpler? (and in line with for instance the GL handling) but someone else should chime in.

in all cases, as currently implemented, declaring an ofSoundStream in .h creates double instance and possible confusion between the global and the declared. not declaring it but using ofSoundStream*() works transparently without ambiguity.
